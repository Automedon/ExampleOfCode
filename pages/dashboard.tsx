import React from "react";
import withApollo from "../lib/withApollo";
import DashBoardList from "../components/Dashboard/DashBoardList";
import Link from "next/link";

const Dashboard = () => {
  return (
    <div>
      <div style={{ textAlign: "center" }}>
        <Link href="/">
          <a>Home</a>
        </Link>
      </div>
      <h1 style={{ textAlign: "center" }}>Automedon CarYard</h1>
      <DashBoardList />
    </div>
  );
};

export const TestingDashboard = Dashboard;
export default withApollo(Dashboard);
