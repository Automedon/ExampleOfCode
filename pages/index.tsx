import React from "react";
import withApollo from "../lib/withApollo";
import Link from "next/link";
import { NextPageContext } from "next";
import { checkLoginServerRender } from "../lib/serverFunctions";
import { Button } from "@material-ui/core";
import Layout from "../components/Layout/Layout";

const Index = () => {
  return (
    <div>
      <p>A simple example repo</p>
      <Button variant="outlined" color="primary">
        <Link href="/register">
          <a>Register</a>
        </Link>
      </Button>
      <br />
      <Button variant="outlined" color="primary">
        <Link href="/login">
          <a>Login</a>
        </Link>
      </Button>
    </div>
  );
};

Index.getInitialProps = (ctx: NextPageContext) => {
  checkLoginServerRender(ctx);
};

export const TestingIndex = Index;

export default withApollo(Index);
