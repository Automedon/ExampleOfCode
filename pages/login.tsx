import React from "react";
import LoginForm from "../components/LoginForm/LoginForm";
import Layout from "../components/Layout/Layout";
import FindUser from "../components/FindUser/FindUser";
import withApollo from "../lib/withApollo";
import Link from "next/link";
import { Button } from "@material-ui/core";

const LoginPage = () => {
  return (
    <Layout>
      <Button variant="outlined" color="primary">
        <Link href="/dashboard">
          <a>DashBoard (You can go here without login but...)</a>
        </Link>
      </Button>
      <LoginForm />
      <FindUser />
    </Layout>
  );
};

export const TestingLoginPage = LoginPage;

export default withApollo(LoginPage);
