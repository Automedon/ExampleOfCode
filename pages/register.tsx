import React from "react";
import RegisterForm from "../components/RegisterForm/RegisterForm";
import Layout from "../components/Layout/Layout";

const RegisterPage = () => {
  return (
    <Layout>
      <RegisterForm />
    </Layout>
  );
};

export default RegisterPage;
