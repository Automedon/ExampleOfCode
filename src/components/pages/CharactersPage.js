import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { charactersSelector } from "../../reducers/characters";
import { connect } from "react-redux";

const CharactersPage = ({ characters }) => (
  <div>
    {characters.length === 0 && (
      <div className="text-center">
        <div className="alert alert-info">
          You have no characters yet. How about creating one?
        </div>
        <Link to="/characters/new" className="btn btn-primary btn-lg">
          Create New Character
        </Link>
      </div>
    )}
  </div>
);

CharactersPage.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.object).isRequired
};

function mapStateToProps(state) {
  return {
    characters: charactersSelector(state)
  };
}

export default connect(mapStateToProps)(CharactersPage);
