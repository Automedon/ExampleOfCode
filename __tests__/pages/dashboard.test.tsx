import renderer from "react-test-renderer";
import React from "react";
import { TestingDashboard } from "../../pages/dashboard";

describe("Snapshot TestingDashboard page", () => {
  it("TestingDashboard renders correctly", () => {
    const tree = renderer.create(<TestingDashboard />);
    expect(tree).toMatchSnapshot();
  });
});
