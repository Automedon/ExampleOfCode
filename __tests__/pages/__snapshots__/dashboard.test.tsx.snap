// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot TestingDashboard page TestingDashboard renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Home
    </a>
  </div>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Automedon CarYard
  </h1>
  <div>
    Loading...
  </div>
</div>
`;

exports[`Snapshot dashboard page dashboard renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Home
    </a>
  </div>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Automedon CarYard
  </h1>
  <div>
    Loading...
  </div>
</div>
`;
